pkg/server/v3/server_test.go

```diff
@@ -578,6 +839,62 @@ func TestAggregatedHandlers(t *testing.T) {
 	}
 }
 
+func TestDeltaAggregatedHandlers(t *testing.T) {
+	config := makeMockConfigWatcher()
+	config.deltaResponses = makeDeltaResponses()
+	resp := makeMockDeltaStream(t)
+
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:    node,
+		TypeUrl: rsrc.ListenerType,
+	}
+	// Delta compress node
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:    node,
+		TypeUrl: rsrc.ClusterType,
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		Node:                   node,
+		TypeUrl:                rsrc.EndpointType,
+		ResourceNamesSubscribe: []string{clusterName},
+	}
+	resp.recv <- &discovery.DeltaDiscoveryRequest{
+		TypeUrl:                rsrc.RouteType,
+		ResourceNamesSubscribe: []string{routeName},
+	}
+
+	s := server.NewServer(context.Background(), config, server.CallbackFuncs{})
+	go func() {
+		if err := s.DeltaAggregatedResources(resp); err != nil {
+			t.Errorf("DeltaAggregatedResources() => got %v, want no error", err)
+		}
+	}()
+
+	count := 0
+	for {
+		select {
+		case <-resp.sent:
+			count++
+			if count >= 4 {
+				close(resp.recv)
+				if want := map[string]int{
+					rsrc.EndpointType: 1,
+					rsrc.ClusterType:  1,
+					rsrc.RouteType:    1,
+					rsrc.ListenerType: 1,
+				}; !reflect.DeepEqual(want, config.counts) {
+					t.Errorf("watch counts => got %v, want %v", config.counts, want)
+				}
+
+				// got all messages
```

Comment:
?

Reply:
That comment isn't very useful. It's essentially saying we can exit since we've successfully retrieved all the messages from ADS. I'll remove